src/analysis/convex/basic.lean

```diff
@@ -117,12 +118,102 @@ end
 lemma segment_translate_preimage (a b c : E) : (λ x, a + x) ⁻¹' [a + b, a + c] = [b, c] :=
 set.ext $ λ x, mem_segment_translate a
 
-lemma segment_translate_image (a b c: E) : (λx, a + x) '' [b, c] = [a + b, a + c] :=
+lemma segment_translate_image (a b c : E) : (λx, a + x) '' [b, c] = [a + b, a + c] :=
 segment_translate_preimage a b c ▸ image_preimage_eq _ $ add_left_surjective a
 
 lemma segment_image (f : E →ₗ[ℝ] F) (a b : E) : f '' [a, b] = [f a, f b] :=
 set.ext (λ x, by simp [segment_eq_image])
 
+/-- Open segment in a vector space. Note that `open_segment x x = {x}` instead of being `∅`. -/
+def open_segment (x y : E) :
+  set E :=
+{z : E | ∃ (a b : ℝ) (ha : 0 < a) (hb : 0 < b) (hab : a + b = 1), a • x + b • y = z}
+
+local notation `]`x `, ` y `[` := open_segment x y
```

Comment:
You introduce the notation, but you don't use it below.
Btw, I wouldn't be surprised if Lean would get thoroughly confused by this notation. It certainly confuses me on paper :stuck_out_tongue_winking_eye:

Reply:
Actually, I ran into problems because Lean decides to first find the [] in stuff like `]a + b, a + c[ = ]b, c[`! We can surely delete it without regret.

Reply:
> You introduce the notation, but you don't use it below.
> Btw, I wouldn't be surprised if Lean would get thoroughly confused by this notation. It certainly confuses me on paper stuck_out_tongue_winking_eye

Lean will certainly be confused, but you shouldn't. This is really one of the greatest notations of mathematics. I have no idea why it's used only in France. It is so much better than `(a, b)` which is impossible to distinguish from the pair notation.

Reply:
I know they use it for tubes in p-adic geometry. If you start taking fibre products of tubes over tubes, I find that the notation becomes extremely confusing. Especially if you don't take care with the typography. Because by default TeX will add extra spacing on the wrong side of the bracket.